src/app/applications/multicast/multicast.service.ts

```diff
@@ -0,0 +1,95 @@
+import { Injectable } from '@angular/core';
+import { UserMinimalService } from '@app/admin/users/user-minimal.service';
+import { RestService } from '@shared/services/rest.service';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { MulticastResponse } from './multicast-response.model';
+import { Multicast, MulticastData } from './multicast.model';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class MulticastService {
+  constructor(
+    private restService: RestService,
+    private userMinimalService: UserMinimalService
+  ) {}
+
+  private multicastURL = 'multicast'; // api endpoint
+  getMulticastsByApplicationId(
+    limit: number,
+    offset: number,
+    applicationId: number
+  ): Observable<MulticastData> {
+    const body = {
+      limit,
+      offset,
+      applicationId,
+    };
+    return this.restService.get(this.multicastURL, body); // get's the multicasts from specific application by the url and the body with applicationId.
+  }
+  get(id: number): Observable<Multicast> {
+    // Get's a single multicast by id.
+    return this.restService.get(this.multicastURL, {}, id).pipe(
+      map((response: MulticastResponse) => {
+        const multicast = this.mapToMulticast(response); // maps the response from backend.
+        return multicast;
+      })
+    );
+  }
+  delete(id: number) {
+    // deletes a chosen multicast by id
+    return this.restService.delete(this.multicastURL, id);
+  }
+  update(multicast: Multicast): Observable<Multicast> {
+    // updates the chosen multicast by id
+    return this.restService
+      .put(this.multicastURL, multicast, multicast.id)
+      .pipe
+      // map((response: MulticastResponse) => {
```

Comment:
Personally, I would say that out-commented code should be deleted. It's not maintained, it will bloat the project and they can be pulled from Git if needed. It should only stay if it's informative enough (and it can't be replaced by a comment). Ex. "this line is a common mistake because X"